### PR TITLE
CLN: GH29547 change string formatting with f-strings (6 files changed)

### PR DESCRIPTION
**Pull request by MatteoFelici**
Replace old string formatting syntax with f-strings #29547

- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Changed 6 files:
- pandas/tests/io/parser/test_header.py
-  pandas/tests/io/test_sql.py
- pandas/tests/io/test_html.py
- pandas/tests/reductions/test_reductions.py
- pandas/tests/reshape/test_melt.py
- pandas/tests/scalar/timedelta/test_timedelta.py

### CODE REVIEW

File: pandas/tests/io/test_html.py [hunk 1]
```diff
@@ -507,7 +507,8 @@ def test_tfoot_read(self):
         Make sure that read_html reads tfoot, containing td or th.
         Ignores empty tfoot
         """
-        data_template = """<table>
+        data_template = (
```
Comment (TomAugspurger) [inline pandas/tests/io/test_html.py:510]: This doesn't seem like an improvement.

File: pandas/tests/reshape/test_melt.py [hunk 2]
```diff
@@ -313,32 +313,30 @@ def test_melt_missing_columns_raises(self):
         df = pd.DataFrame(np.random.randn(5, 4), columns=list("abcd"))
 
         # Try to melt with missing `value_vars` column name
-        msg = "The following '{Var}' are not present in the DataFrame: {Col}"
-        with pytest.raises(
-            KeyError, match=msg.format(Var="value_vars", Col="\\['C'\\]")
-        ):
+        msg = (
```
Comment (TomAugspurger) [inline pandas/tests/reshape/test_melt.py:316]: Can you revert this batch?

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 3]
```diff
@@ -83,12 +83,12 @@ def check(value):
         assert rng.microseconds == 0
         assert rng.nanoseconds == 0
 
-        msg = "'Timedelta' object has no attribute '{}'"
-        with pytest.raises(AttributeError, match=msg.format("hours")):
+        msg = lambda x: f"'Timedelta' object has no attribute '{x}'"
```
Comment (TomAugspurger) [inline pandas/tests/scalar/timedelta/test_timedelta.py:86]: Also revert these.

**Review by TomAugspurger — COMMENTED** [review]
Thanks. Left a few comments on places where the f-strings aren't an improvement.

File: pandas/tests/io/test_html.py (hunk 1 above)
Comment (MatteoFelici) [reply pandas/tests/io/test_html.py]: reverted

File: pandas/tests/scalar/timedelta/test_timedelta.py (hunk 3 above)
Comment (MatteoFelici) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: reverted

File: pandas/tests/reshape/test_melt.py (hunk 2 above)
Comment (MatteoFelici) [reply pandas/tests/reshape/test_melt.py]: reverted

**Comment by jreback** [comment]
thanks @MatteoFelici

**Comment by MatteoFelici** [comment]
@TomAugspurger I was looking at other PR converting `msg` object into functions, in order to remove the f-string, like https://github.com/pandas-dev/pandas/pull/34399. You suggested reverting those modifications. Should not we take a decision about this and be consistent with all modifications?

**Pull request merged**